### PR TITLE
Config option to turn off spawn chunk persistance

### PR DESCRIPTION
**Pull request by mastercoms**
This commit will give the option to turn off keeping spawn loaded in the config.
world:
keep-spawn-chunks: false

That would not persist any spawn chunks. It is true by default, though.

### CODE REVIEW

**Comment by SpaceManiac** [comment]
Is there an advantage to including this in core over letting a plugin do it? (Multiverse, for example, supports it).

File: src/main/java/net/glowstone/util/ServerConfig.java [hunk 1]
```diff
@@ -307,6 +307,7 @@ private boolean migrate() {
         GENERATOR_SETTINGS("world.gen-settings", "", Migrate.PROPS, "generator-settings"),
         ALLOW_NETHER("world.allow-nether", true, Migrate.PROPS, "allow-nether"),
         ALLOW_END("world.allow-end", true, Migrate.BUKKIT, "settings.allow-end"),
+        PERSIST_SPAWN("world.keep-spawn-chunks", true),
```
Comment (turt2live) [inline src/main/java/net/glowstone/util/ServerConfig.java]: Maybe use "keep spawn loaded" or similar. This currently implies that the chunks will be kept across worlds (someone will ask it eventually)

**Comment by mastercoms** [comment]
@SpaceManiac In future coding, we might need to know more information about spawn and it would be preferable to use definitive server code rather than plugin code.
Also, it allows server owners to tweak these settings without adding extra bloat like Multiverse would. Sometimes people want to configure the spawn persistence but don't want a new world manager.

File: src/main/java/net/glowstone/util/ServerConfig.java (hunk 1 above)
Comment (mastercoms) [inline src/main/java/net/glowstone/util/ServerConfig.java]: @turt2live Will do.

**Comment by turt2live** [comment]
@mastercoms This PR is currently not mergable. Please fix that ASAP!

**Comment by turt2live** [comment]
@mastercoms still has merge conflicts :(

**Comment by mastercoms** [comment]
@turt2live Ok, I'll have time over the weekend to fix it. Sorry for the delay!

**Comment by turt2live** [comment]
@mastercoms Please update your PR as soon as possible.

**Comment by mastercoms** [comment]
@turt2live I will make a new PR for it.

**Pull request closed without merging**